packages/block-editor/src/components/block-icon/index.native.js

```diff
@@ -8,46 +8,40 @@ import { View } from 'react-native';
  */
 import { Icon } from '@wordpress/components';
 import { blockDefault } from '@wordpress/icons';
-import { withPreferredColorScheme } from '@wordpress/compose';
+import { usePreferredColorSchemeStyle } from '@wordpress/compose';
 
 /**
  * Internal dependencies
  */
 import styles from './style.scss';
 
-export function BlockIcon( {
-	icon,
-	fill,
-	size,
-	showColors = false,
-	getStylesFromColorScheme,
-} ) {
+export function BlockIcon( { icon, fill, size, showColors = false } ) {
 	if ( icon?.src === 'block-default' ) {
 		icon = {
 			src: blockDefault,
 		};
 	}
 
+	const defaultFill = usePreferredColorSchemeStyle(
+		styles.iconPlaceholder,
+		styles.iconPlaceholderDark
+	)?.fill;
+	const iconForeground = showColors ? icon?.foreground : undefined;
+
 	const renderedIcon = (
 		<Icon
 			icon={ icon && icon.src ? icon.src : icon }
-			{ ...( fill && { fill } ) }
+			fill={ fill || iconForeground || defaultFill }
```

Comment:
I decided to apply a small refactor for calculating the fill color, now it includes the foreground icon (more info in [this comment](https://github.com/WordPress/gutenberg/pull/34967/files#r712868424)) and the default color (more info in [this comment](https://github.com/WordPress/gutenberg/pull/34967/files#r712874556)).